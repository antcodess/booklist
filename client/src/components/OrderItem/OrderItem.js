import React from 'react';
import { withRouter } from 'react-router-dom';

import './OrderItem.scss';

const OrderItem = ({ item, history }) => {
  const { routeName, imageUrl, title, quantity, price } = item;

  return (
    <div className='order-item'> 
      <div className='image-container' onClick={() => history.push(`/${routeName}`)}>
        <img src={imageUrl} alt='book' /> 
      </div>
      <span className='title' onClick={() => history.push(`/${routeName}`)}>{title}</span>
      <span className='quantity'>{quantity}</span>
      <span className='price'>${price}</span>
    </div>
  )
};

export default withRouter(OrderItem);