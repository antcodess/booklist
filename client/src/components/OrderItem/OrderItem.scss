.order-item {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 10px;
  margin-top: 40px;
  text-align: center;

  .image-container {
    cursor: pointer;

    img {
      width: 100%;
      height: auto;
    }
  }
  .title,
  .quantity,
  .price,
  .date {
    margin: auto 0;
  }

  .title {
    cursor: pointer;
  }
}