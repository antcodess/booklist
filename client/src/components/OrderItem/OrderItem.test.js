import React from 'react';
import { shallow } from 'enzyme';
import { OrderItem } from './OrderItem';

describe('OrderItem Component', () => {
  let wrapper;
  let mockItem;
  let mockHistory;
  let dummyFunction = jest.fn();
  
  beforeEach(() => {

    mockItem = {
      title: 'Item 1', 
      imageUrl: 'example.com', 
      quantity: 1, 
      price: 20, 
      routeName: '/products/item-1'
    };
    
    mockHistory = { push: jest.fn() };

    const mockProps = {
      item: mockItem,
      history: mockHistory
    };
    
    wrapper = shallow(<OrderItem { ...mockProps } />);
  });

  it('should render OrderItem Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push when clicked for image-container div', () => {
    wrapper.find('.image-container').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it('should call history.push when clicked for title div', () => {
    wrapper.find('.title').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });
});
