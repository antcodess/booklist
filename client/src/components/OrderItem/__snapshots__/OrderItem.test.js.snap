// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderItem Component should render OrderItem Component 1`] = `ShallowWrapper {}`;
